### DISCUSSION TITLE
Question: TypeScript support for gatsby-* files?

### DISCUSSION THREAD

**Discussion by mrseanbaines** [gatsbyjs/gatsby#26314] [Help] [answered]
## Summary
Are there any plans to natively support TypeScript for `gatsby-*` files? I see there was a similar question [here](https://github.com/gatsbyjs/gatsby/issues/1457), but that's about 3 years old and TypeScript is now a first class citizen in Gatsby so I wonder if this is in the roadmap now?

If not (or until such a day), is there a suggested way to achieve the same effect? Right now I'm registering `ts-node` in `gatsby-config.js` and importing/exporting the gatsby config from a separate TypeScript file:

```js
// gatsby-config.js

require('ts-node').register()

module.exports = require('./src/config/gatsby-config.ts')
```

```ts
// src/config/gatsby-config.ts

import { GatsbyConfig } from 'gatsby'

const gatsbyConfig: GatsbyConfig = {
  siteMetadata: {
    siteName: 'My Cool Site',
  },
}

export default gatsbyConfig
```

**Comment by blainekasten** [comment, marked as answer]
@mrseanbaines It absolutely *is* on the roadmap! Though it's hard to pin down when it'll be completed. In the meantime, your approach looks very reasonable!

**Comment by mrseanbaines** [comment]
Good to hear! And good to know that my approach for now looks okay. Thanks for the response 💥

**Comment by muescha** [comment]
@blainekasten until it is implemented, maybe the workaround can be placed in docs? 

for example here: https://www.gatsbyjs.com/docs/typescript/

**Comment by blainekasten** [comment]
@muescha that is a neat idea! But I think our time is better spent getting the code to work for it :)
